lib/src/lifecycle_module.dart

```diff
@@ -801,6 +801,7 @@ abstract class LifecycleModule extends SimpleModule
       rethrow;
     } catch (error, stackTrace) {
       _didUnloadController.addError(error, stackTrace);
+      await _disposableProxy.dispose();
```

Comment:
I would be concerned that the call to `_disposableProxy.dispose()` above was the source of the exception that is being caught in this bock. A second attempt to call it would *probably* result in another exception preventing `_postUnlaodDisposable.dispose()` from being called.